ansible/library/conn_graph_facts.py

```diff
@@ -231,28 +228,63 @@ def get_host_connections(self, hostname):
         else:
             return self.links
 
+    def contains_hosts(self, hostnames):
+        return set(hostnames) <= set(self.devices)
+
+
+LAB_CONNECTION_GRAPH_FILE = 'graph_files.yml'
+EMPTY_GRAPH_FILE = 'empty_graph.xml'
+LAB_GRAPHFILE_PATH = 'files/'
+
+"""
+    Find a graph file contains all devices in testbed.
+"""
+def find_graph(hostnames, anchor):
+    filename = LAB_GRAPHFILE_PATH + LAB_CONNECTION_GRAPH_FILE
+    with open(filename) as fd:
+        file_list = yaml.load(fd)
+
+    for fn in file_list:
+        filename = LAB_GRAPHFILE_PATH + fn
+        lab_graph = Parse_Lab_Graph(filename)
+        lab_graph.parse_graph()
+        if hostnames and lab_graph.contains_hosts(hostnames):
+            return lab_graph
+        if anchor and lab_graph.contains_hosts(anchor):
```

Comment:
Maybe this is explained somewhere but what is `anchor`?

Reply:
Explained in the PR comments.

Reply:
The naming is still a little confusing to me, but this can be iterated on.